### PR TITLE
drivers: nxp: Add GAU drivers for DAC and ADC

### PR DESCRIPTION
**Pull request by decsny**
Add driver for DAC and ADC present on RW SOCs which are part of the GAU (General Analog Unit) module

### CODE REVIEW

**Comment by dleach02** [comment]
update west manifiest and address merge conflicts

**Comment by dleach02** [comment]
@decsny, please address the merge conflicts

File: dts/bindings/adc/nxp,gau-adc.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    enum:
+      - "full-bias"
+      - "half-bias"
+    default: "full-bias"
+
+  nxp,trigger-source:
+    type: string
+    description: |
+      Conversion trigger source.
+      Default is "software" which is the only value currently supported".
```
Comment (danieldegrasse) [inline dts/bindings/adc/nxp,gau-adc.yaml:41]:
```suggestion
      Default is "software" which is the only value currently supported.
```

File: dts/bindings/adc/nxp,gau-adc.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    enum:
+      - "gpt"
+      - "acomp"
+      - "gpio40"
```
Comment (danieldegrasse) [inline dts/bindings/adc/nxp,gau-adc.yaml:45]: These values look like they will be specific to the RW61x SOC. I guess we only have that reference manual to go on though for these names. Any thoughts here on what we could call these that would be more generic?

Perhaps we should simply direct users to the reference manual for the meanings of each value?

File: drivers/adc/adc_mcux_gau_adc.c [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+static int mcux_gau_adc_do_read(const struct device *dev,
+		   const struct adc_sequence *sequence)
+{
+	const struct mcux_gau_adc_config *config = dev->config;
+	ADC_Type *base = config->base;
+	struct mcux_gau_adc_data *data = dev->data;
+	uint8_t num_channels = 0;
+
+	/* if user selected channe >= NUM_ADC_CHANNELS that is invalid */
```
Comment (danieldegrasse) [inline drivers/adc/adc_mcux_gau_adc.c:200]:
```suggestion
	/* if user selected channel >= NUM_ADC_CHANNELS that is invalid */
```

File: drivers/adc/adc_mcux_gau_adc.c [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+#include <zephyr/irq.h>
+#include <errno.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(adc_mcux_gau_adc, CONFIG_ADC_LOG_LEVEL);
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#include "fsl_adc.h"
```
Comment (danieldegrasse) [inline drivers/adc/adc_mcux_gau_adc.c:19]:
```suggestion
#include <fsl_adc.h>
```

File: drivers/dac/Kconfig.mcux [hunk 5]
```diff
@@ -31,3 +31,10 @@ config DAC_MCUX_DAC32_TESTOUT
 	depends on DAC_MCUX_DAC32
 	help
 	  Enable the DAC test output.
+
+config DAC_MCUX_GAU
+	bool "NXP GAU DAC driver"
+	default y
+	depends on DT_HAS_NXP_GAU_DAC_ENABLED
+	help
+	  Enable the driver for the NXP GAU D
```
Comment (danieldegrasse) [inline drivers/dac/Kconfig.mcux:40]:
```suggestion
	  Enable the driver for the NXP GAU DAC
```

File: drivers/dac/dac_mcux_gau.c [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_gau_dac
+
+#include <zephyr/drivers/dac.h>
+
+#include "fsl_dac.h"
```
Comment (danieldegrasse) [inline drivers/dac/dac_mcux_gau.c:11]:
```suggestion
#include <fsl_dac.h>
```

File: dts/bindings/adc/nxp,gau-adc.yaml (hunk 2 above)
Comment (decsny) [reply dts/bindings/adc/nxp,gau-adc.yaml:45]: Yeah I'm actually just going to delete this property

File: drivers/dac/Kconfig.mcux (hunk 5 above)
Comment (martinjaeger) [inline drivers/dac/Kconfig.mcux:40]: Did you mean to write this?

```suggestion
	  Enable the driver for the NXP GAU DAC
```

**Review by martinjaeger — CHANGES_REQUESTED** [review]
DAC looks good to me. Only found a small typo.

File: drivers/dac/Kconfig.mcux (hunk 5 above)
Comment (decsny) [reply drivers/dac/Kconfig.mcux]: ugh could have sworn I fixed this, daniel commented the same thing, thanks

**Comment by decsny** [comment]
@sylvioalves there is some esp32 dev kit failing ci on this PR, can you check about it, I think it's unrelated

**Comment by sylvioalves** [comment]
> @sylvioalves there is some esp32 dev kit failing ci on this PR, can you check about it, I think it's unrelated

`/zephyr/zephyr/tests/drivers/adc/adc_api/src/test_adc.c:45:2: error: #error "Unsupported board."` We can fix it.

**Review by danieldegrasse — DISMISSED** [review]

**Review by martinjaeger — DISMISSED** [review]

**Comment by decsny** [comment]
@anangl  please help review

**Comment by decsny** [comment]
added dev review because assignee not responsive

File: drivers/dac/dac_mcux_gau.c [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+{
+	const struct nxp_gau_dac_config *config = dev->config;
+
+	DAC_SetChannelData(config->base,
+			(uint32_t)convert_channel_id(channel),
+			(uint16_t)value);
+	return 0;
+};
+
+struct dac_driver_api nxp_gau_dac_driver_api = {
```
Comment (anangl) [inline drivers/dac/dac_mcux_gau.c:78]: Why not `static const`?

**Review by anangl — DISMISSED** [review]

File: drivers/dac/dac_mcux_gau.c (hunk 7 above)
Comment (decsny) [reply drivers/dac/dac_mcux_gau.c]: it's a mistake, I'll fix it

**Review by anangl — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**